cinema/views.py

```diff
@@ -24,28 +27,60 @@
 class GenreViewSet(viewsets.ModelViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self) -> list:
+        if self.action in ("list", "create"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        else:
+            raise NotFound("Method not allowed")
+        return [permission() for permission in permission_classes]
 
 
 class ActorViewSet(viewsets.ModelViewSet):
     queryset = Actor.objects.all()
     serializer_class = ActorSerializer
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self) -> list:
+        if self.action in ("list", "create"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        else:
+            raise NotFound("Method not allowed")
```

Comment:
Similar to the GenreViewSet, the `raise NotFound("Method not allowed")` statement should be replaced with `MethodNotAllowed` to correctly indicate that the method is not allowed.